src/controller/eme-controller.js

```diff
@@ -14,78 +14,79 @@ const { XMLHttpRequest } = window;
+  }
+
+  if (drmSystemOptions.initDataTypes) {
+    baseConfig.initDataTypes = drmSystemOptions.initDataTypes;
+  }
+
+  if (drmSystemOptions.label) {
+    baseConfig.label = drmSystemOptions.label;
+  }
+
+  if (drmSystemOptions.persistentState) {
+    baseConfig.persistentState = drmSystemOptions.persistentState;
+  }
+
+  if (drmSystemOptions.sessionTypes) {
+    baseConfig.sessionTypes = drmSystemOptions.sessionTypes;
+  }
+
   audioCodecs.forEach((codec) => {
-    baseConfig.audioCapabilities.push({
-      contentType: `audio/mp4; codecs="${codec}"`,
-      robustness: drmSystemOptions.audioRobustness || ''
-    });
+    const audioCapability = {
+      contentType: `audio/mp4; codecs="${codec}"`
+    };
+    if (drmSystemOptions.audioRobustness) {
+      audioCapability.robustness = drmSystemOptions.audioRobustness;
+    }
+    baseConfig.audioCapabilities.push(audioCapability);
   });
   videoCodecs.forEach((codec) => {
-    baseConfig.videoCapabilities.push({
-      contentType: `video/mp4; codecs="${codec}"`,
-      robustness: drmSystemOptions.videoRobustness || ''
-    });
+    const videoCapability = {
+      contentType: `video/mp4; codecs="${codec}"`
+    };
+    if (drmSystemOptions.videoRobustness) {
+      videoCapability.robustness = drmSystemOptions.videoRobustness;
+    }
+    baseConfig.videoCapabilities.push(videoCapability);
   });
 
   return [
     baseConfig
   ];
 };
 
-const createPlayreadyMediaKeySystemConfigurations = function (audioCodecs, videoCodecs, drmSystemOptions = {}) { /* jshint ignore:line */
-  const baseConfig = {
-    // initDataTypes: ['keyids', 'mp4'],
-    // label: "",
-    // persistentState: "not-allowed", // or "required" ?
-    // distinctiveIdentifier: "not-allowed", // or "required" ?
-    // sessionTypes: ['temporary'],
-    audioCapabilities: [
-      // { contentType: 'audio/mp4; codecs="mp4a.40.2"' }
-    ],
-    videoCapabilities: [
-      // { contentType: 'video/mp4; codecs="avc1.42E01E"' }
-    ]
-  };
+/**
+ * @see https://developer.mozilla.org/en-US/docs/Web/API/MediaKeySystemConfiguration
+ * @param {Array<string>} audioCodecs List of required audio codecs to support
+ * @param {Array<string>} videoCodecs List of required video codecs to support
+ * @param {object} drmSystemOptions Optional parameters/requirements for the key-system
+ * @returns {Array<MediaSystemConfiguration>} An array of supported configurations
+ */
 
-  audioCodecs.forEach((codec) => {
-    baseConfig.audioCapabilities.push({
-      contentType: `audio/mp4; codecs="${codec}"`,
-      robustness: drmSystemOptions.audioRobustness || ''
-    });
-  });
-  videoCodecs.forEach((codec) => {
-    baseConfig.videoCapabilities.push({
-      contentType: `video/mp4; codecs="${codec}"`,
-      robustness: drmSystemOptions.videoRobustness || ''
-    });
-  });
+const createWidevineMediaKeySystemConfigurations = function (audioCodecs, videoCodecs, drmSystemOptions = {}) { /* jshint ignore:line */
```

Comment:
should we remove these methods and directly use `buildKeySystemConfigurations` in https://github.com/Tubitv/hls.js/pull/8/files#diff-1cb8b36ed42a658ab63ac970ab1bd648d67f9b199f95fd279f8a50c8d51df2a6R107?

Reply:
+1, it has no differences between these 2 key systems

Reply:
Done